gnuradio-satellites 4.0.0

daniestevez opened the pull request:
This is essentially the same as #17 sent by the bot, but I have added a commit to up the versions of GNU Radio and Volk in the CI to 3.9.0 and 2.4.1 respectively. The main goal of this PR is to see if this is enough to get the pipeline building correctly.

I'm leaving the checklist below that comes with the PR template, but I think all that has been taken care of by the bot.

<!--
Thank you for pull request.
Below are a few things we ask you kindly to self-check before getting a review. Remove checks that are not relevant.
-->
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

<!--
Please note any issues this fixes using [closing keywords]( https://help.github.com/articles/closing-issues-using-keywords/ ):
-->

<!--
Please add any other relevant info below:
-->

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipe```) and found it was in an excellent condition.

daniestevez commented:
This is still failing. Apparently the problem now is that pybind11 is missing from the CI container. I'll get to it.

daniestevez commented:
Linux and OSX builds are now working.

Windows builds are failing with the following problem while running `cmake`:
```
CMake Error in lib/CMakeLists.txt:
  Imported target "gnuradio::gnuradio-runtime" includes non-existent path

    "D:\bld\gnuradio_1614958732532\_h_env\lib\site-packages\numpy\core\include"

  in its INTERFACE_INCLUDE_DIRECTORIES.  Possible reasons include:

  * The path was deleted, renamed, or moved to another location.

  * An install or uninstall procedure did not complete successfully.

  * The installation package was faulty and references files it does not
  provide.



CMake Error in lib/CMakeLists.txt:
  Imported target "gnuradio::gnuradio-runtime" includes non-existent path

    "D:\bld\gnuradio_1614958732532\_h_env\lib\site-packages\numpy\core\include"

  in its INTERFACE_INCLUDE_DIRECTORIES.  Possible reasons include:

  * The path was deleted, renamed, or moved to another location.

  * An install or uninstall procedure did not complete successfully.

  * The installation package was faulty and references files it does not
  provide.
```

I am a bit clueless about this problem. One thing I'm going to do is to revert 0787fe9 since it seems to me that perhaps it isn't necessary to list pybind11 in `.ci_support`.

ryanvolz commented:
Hey @daniestevez! `gnuradio-satellites` is the first of the OOTs on conda-forge to actually have a release that supports 3.9, so unfortunately that means it's something of a guinea pig. The fact that it's trying to build against GNU Radio 3.8 is expected, because that is the globally pinned version for conda-forge (everything builds against that for consistency). I'll plan to update that pin to 3.9 once more things support it, but until then we'll need manual intervention for `gnuradio-satellites` like you've already done.

The error on Windows is more concerning, and I think it probably points to a bit of a packaging problem with GNU Radio 3.9 where the build prefix is mistakenly not overwritten. I'll have to look into it a bit more.

By the way, we can easily add you as a maintainer for `gnuradio-satellites` on conda-forge if you would like. That would give you commit access and, I guess, the ability to push to bot branches so you don't have to make your own PR in these cases. No problem if you'd rather not, I just thought I'd offer.

ryanvolz commented:
> The error on Windows is more concerning, and I think it probably points to a bit of a packaging problem with GNU Radio 3.9 where the build prefix is mistakenly not overwritten. I'll have to look into it a bit more.

As I thought, GR's CMake is including the build prefix path and it's not getting overwritten by conda as it should. It's happening because the path appears in that file twice, but with different styles of path separators, and for some reason conda can only handle one string for prefix replacement per file:
```
INTERFACE_INCLUDE_DIRECTORIES "D:\\bld\\gnuradio_1614958604036\\_h_env\\lib\\site-packages\\numpy\\core\\include;D:/bld/gnuradio_1614958604036/_h_env/Library/include;${_IMPORT_PREFIX}/include"
```
This is going to require a patch and re-build of GNU Radio 3.9.

daniestevez commented:
Hi Ryan,

Thanks for looking into this.

Yes, please add me as a maintainer to the repo. It makes sense that I am maintainer, since I might sometimes help with things like this. I'll be careful not to break anything ;-)

ryanvolz commented on .ci_support/linux_64_python3.6.____cpython.yaml:
```diff
@@ -33,7 +33,7 @@ python:
 - linux-64
 volk:
-- '2.4'
+- '2.4.1'
```
It isn't necessary to add the ".1" here. It will build against the latest version of 2.4. But for now it doesn't hurt.

The way these .ci_support files work is that they are generated by `conda-smithy` whenever the recipe is re-rendered. The particular values for these pins come from the global pinning file installed by `conda-forge-pinning`. They get updated with PRs to that feedstock, and then there are bots that handle PRs to re-render any affected recipes. But since we're not ready to globally update to gnuradio_core 3.9 yet, updating these manually is perfectly fine. Just know that they will get overwritten whenever the recipe is re-rendered.

daniestevez replied:
Thanks for the explanation. I have zero experience about how these things work. I guess we can leave it as '2.4.1' if it's going to be overwritten eventually.

ryanvolz reviewed (commented):
To get added as a maintainer, you can simply add your github handle to the list at the bottom of the `meta.yaml` file. When the PR is merged, it should trigger a bot that will add you to the conda-forge organization and give you commit rights to this repository.

ryanvolz commented:
I've patched GNU Radio to hopefully fix the Windows build here: https://github.com/conda-forge/gnuradio-feedstock/pull/45. Once I confirm that the problem is fixed and all of the packages get rebuilt (it'll take a few hours), then you can use another commit here to trigger a re-build. I'll add an automerge label so that it will be merged once the tests pass.

daniestevez commented:
> To get added as a maintainer, you can simply add your github handle to the list at the bottom of the `meta.yaml` file. When the PR is merged, it should trigger a bot that will add you to the conda-forge organization and give you commit rights to this repository.

Perfect. I've just pushed a commit with that change to this PR.

daniestevez commented:
I've waited until conda-forge/gnuradio-feedstock#45 built and was merged successfully, and then triggered rebuild of this PR. The build for Windows failed again. Comparing the logs with the builds I did earlier, it seems to me that it's using the same gnuradio package as before. For instance, `gnuradio-core 3.9.0.0-py36h478160b_2` for Windows and Python 3.6.

Looking at the [build log](https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=287042&view=logs&jobId=171a126d-c574-5c8c-1269-ff3b989e923d&j=171a126d-c574-5c8c-1269-ff3b989e923d&t=f12b9669-b50e-5b51-73e1-df5038e385ed) of conda-forge/gnuradio-feedstock#45 it seems that the packages were not uploaded:

```
Distribution D:\bld\win-64\gnuradio-core-3.9.0.0-py36h478160b_3.tar.bz2 is new for cf-staging, but no upload is taking place because the BINSTAR_TOKEN/STAGING_BINSTAR_TOKEN is missing or empty.
```

In fact, all the [conda-forge packages for gnuradio](https://anaconda.org/conda-forge/gnuradio/files) are from 2 days ago, so it seems that nothing has been uploaded.

ryanvolz commented:
You're correct that it hasn't been uploaded yet. What was built for the PR was just for testing. It's currently building off the master branch post merge. That'll finish in a couple hours, and then you'll have to allow for 30-60 minutes after the successful build for the packages to fully propagate through anaconda.org's CDN. I know it takes forever with long-building packages like gnuradio.

daniestevez commented:
Thanks @ryanvolz! I've seen that everything has now built correctly, the PR has been merged, and the packages are already in conda-forge.

It seems that the pipeline of the OSX Python3.8 package in the merge commit https://github.com/conda-forge/gnuradio-satellites-feedstock/commit/ae9aa845c970fcf2ed95c98e4fe372ddbe527907 failed during the checkout, so this package hasn't been built. Is there a way to trigger a rebuild for this?

ryanvolz commented:
I was able to click "re-run" from the Github Actions interface to the failing build. It looks like it's going now, so that should take care of it. Otherwise, recommended practice would be to push an empty commit to master.
